graylog2-server/src/main/java/org/graylog2/rest/resources/roles/RolesResource.java

```diff
@@ -0,0 +1,261 @@
+
+        final Role role = roleService.load(name);
+        return RoleResponse.create(role.getName(), Optional.fromNullable(role.getDescription()), role.getPermissions(), role.isReadOnly());
+    }
+
+    @POST
+    @RequiresPermissions(RestPermissions.ROLES_CREATE)
+    @ApiOperation(value = "Create a new role", notes = "")
+    public Response create(@ApiParam(name = "JSON body", value = "The new role to create", required = true) @Valid @NotNull RoleResponse roleResponse) {
+        try {
+            roleService.load(roleResponse.name());
+            throw new BadRequestException("Role " + roleResponse.name() + " already exists.");
+        } catch (NotFoundException ignored) {
+        }
+
+        Role role = new RoleImpl();
+        role.setName(roleResponse.name());
+        role.setPermissions(roleResponse.permissions());
+        role.setDescription(roleResponse.description().orNull());
+
+        try {
+            role = roleService.save(role);
+        } catch (ValidationException e) {
+            log.error("Invalid role creation request.");
+            throw new BadRequestException(e);
+        }
+
+        final URI uri = getUriBuilderToSelf().path(RolesResource.class)
+                .path("{rolename}")
+                .build(role.getName());
+
+        return Response.created(uri).entity(RoleResponse.create(role.getName(),
+                                                                Optional.fromNullable(role.getDescription()),
+                                                                role.getPermissions(),
+                                                                role.isReadOnly())).build();
+    }
+
+    @PUT
+    @Path("{rolename}")
+    @ApiOperation("Update an existing role")
+    public RoleResponse update(
+            @ApiParam(name = "rolename", required = true) @PathParam("rolename") String name,
+            @ApiParam(name = "JSON Body", value = "The new representation of the role", required = true) RoleResponse role) throws NotFoundException {
+        final Role roleToUpdate = roleService.load(name);
+
+        if (roleToUpdate.isReadOnly()) {
+            throw new BadRequestException("Cannot update read only role " + name);
+        }
+        roleToUpdate.setName(role.name());
+        roleToUpdate.setDescription(role.description().orNull());
+        roleToUpdate.setPermissions(role.permissions());
+        try {
+            roleService.save(roleToUpdate);
+        } catch (ValidationException e) {
+            throw new BadRequestException(e);
+        }
+        return RoleResponse.create(roleToUpdate.getName(), Optional.fromNullable(roleToUpdate.getDescription()), roleToUpdate.getPermissions(),
+                                   role.readOnly());
+    }
+
+    @DELETE
+    @Path("{rolename}")
+    @ApiOperation(value = "Remove the named role")
+    public Response delete(@ApiParam(name = "rolename", required = true) @PathParam("rolename") String name) throws NotFoundException {
```

Comment:
Just a remark: methods with return type `void` will automatically generate an HTTP response with status code 204 (No Content). No need to explicitly create such a `Response` object.